eng/Packages/General.props

```diff
@@ -15,10 +15,10 @@
     <PackageVersion Include="Microsoft.Azure.KeyVault.AzureServiceDeploy" Version="3.0.0" />
     <PackageVersion Include="Microsoft.Azure.Services.AppAuthentication" Version="1.6.1" />
     <PackageVersion Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.4" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.Common" Version="4.0.1" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.0.1" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.0.1" />
-    <PackageVersion Include="Microsoft.CodeAnalysis" Version="4.0.1" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.Common" Version="4.4.0-6.23101.13" />
```

Comment:
Do note that doing this will have an impact on the supported VS versions for our source generators, so we should be very sure about making these changes. You can find matching versions support in [this table](https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/extensibility/roslyn-version-support.md), but essentially before this change our source generators would have run fine on VS 17.0, but after this they will require 17.4.